client/src/app/queries/generators.ts

```diff
@@ -0,0 +1,96 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+import { AxiosError } from "axios";
+import { AssetGenerator } from "@app/api/models";
+import {
+  createGenerator,
+  deleteGenerator,
+  getGeneratorById,
+  getGenerators,
+  updateGenerator,
+} from "@app/api/rest";
+
+export const GENERATORS_QUERY_KEY = "generators";
+export const GENERATOR_QUERY_KEY = "generator";
+
+export const useFetchGenerators = () => {
```

Comment:
In #2472, `refetchInterval` args have been added to add useFetch___() hooks.  Let's add it here as well.